Assets/_Developers/Rolo/Scripts/MapRework/MapStationDrag.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class MapStationDrag : MonoBehaviour
+{
+    [SerializeField] private LayerMask RaycastLayers = default(LayerMask);
+
+    Transform stationTransform;
+
+    bool isPressed;
+    // Start is called before the first frame update
+    void Start()
+    {
+        stationTransform = null;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (isPressed)
+        {
+            if (stationTransform != null)
+            {
+                DragStation();
+            }
+        }
+        else if (!isPressed && stationTransform != null)
+        {
+            SnapStationInPlace();
+        }
+    }
+
+    private void SnapStationInPlace()
+    {
+        StationBehaviour sb = stationTransform.gameObject.GetComponent<StationBehaviour>();
+        Transform stationSlot = CheckIfCursorOnMapItem("StationSlot");
+        if (stationSlot != null)
+        {
+            stationTransform.position = stationSlot.position;
+            sb.previousPos = stationTransform.position;
+        }
+        else
+        {
+            stationTransform.position = sb.previousPos;
+        }
+
+        stationTransform = null;
+    }
+
+    private void DragStation()
+    {
+        Vector3 mousePosition = GetCursorPositionInWorld();
+        Vector2 tokenTranslation = mousePosition - stationTransform.position;
```

Comment:
Design Question, when dragging a token, should it snap to the mouse position, or should it remain at the mouse click position?